lib/streamlit/runtime/state/session_state_proxy.py

```diff
@@ -55,6 +59,76 @@ def get_session_state() -> SafeSessionState:
     return ctx.session_state
 
 
+def get_query_params() -> QueryParams:
+    from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+    ctx = get_script_run_ctx()
+
+    if ctx is None:
+        return QueryParams()
+    return get_session_state()._state._query_params
+
+
+class QueryParamsProxy(MutableMapping[Key, Any]):
```

Comment:
Should this exist in another file? It feels like it's not necessary since it's a small amount of code but would be open to having it in another file so it's easier to find(?)